components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.webhook.management.internal.dao.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.webhook.management.api.constant.ErrorMessage;
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtException;
+import org.wso2.carbon.identity.webhook.management.api.model.Webhook;
+import org.wso2.carbon.identity.webhook.management.internal.constant.WebhookSQLConstants.Column;
+import org.wso2.carbon.identity.webhook.management.internal.constant.WebhookSQLConstants.Query;
+import org.wso2.carbon.identity.webhook.management.internal.dao.WebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.util.WebhookManagementExceptionHandler;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Implementation of WebhookManagementDAO interface.
+ */
+public class WebhookManagementDAOImpl implements WebhookManagementDAO {
+
+    private static final String WEBHOOK_SCHEMA_VERSION = "v1";
+    private static final String WEBHOOK_VERSION = "1.0.0";
+
+    /**
+     * Create a new webhook subscription in the database.
+     *
+     * @param webhook  Webhook subscription to be created.
+     * @param tenantId Tenant ID.
+     * @throws WebhookMgtException If an error occurs while creating the webhook subscription.
+     */
+    @Override
+    public void createWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                Integer webhookInternalId = addWebhookToDB(webhook, tenantId);
+                if (webhook.getEventsSubscribed() != null && !webhook.getEventsSubscribed().isEmpty() &&
+                        webhookInternalId != null) {
```

Comment:
So here we seems to check if at least one event is subscribed by the webhook. Ideally this check should happen at service layer, else it will create inconsitency in user experience